src/run-context/runContext.ts

```diff
@@ -8,8 +12,8 @@ export const PCError = {
 };
 
 export class RunContext {
-  private pc: VmPointer;
-  private ap: VmPointer;
+  private pc: MemoryPointer;
+  private ap: MemoryPointer;
   private fp: ProgramCounter;
```

Comment:
Shouldn't ap and fp be `MemoryPointer`?, while pc `ProgramCounter`

Reply:
thx nice catch, fixed!